ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -249,7 +249,7 @@ protected String resolveUserToRunAs(Map<String, String> cache, Table t, Partitio
 
     String user = cache.get(fullTableName);
     if (user == null) {
-      user = findUserToRunAs(sd.getLocation(), t);
+      user = TxnUtils.findUserToRunAs(sd.getLocation(), t, conf);
```

Comment:
Indenependent change?

Reply:
nope, refactor needed to set runAs user for `safe-delete` in AcidEventListener.dropTable()